terminal/src/main/java/org/jline/terminal/impl/AbstractWindowsTerminal.java

```diff
@@ -423,8 +424,11 @@ public boolean canPauseResume() {
     }
 
     @Override
-    public void pause() {
-        paused.set(true);
+    public void pause() throws InterruptedException {
+    	if (paused.compareAndSet(false, true)) {
+            this.pump.interrupt();
```

Comment:
Nothing here will actually throw the interrupt exception. Should we do an `Thread.interrupt()` check at the end of the the thread? Should we even bother interrupting it if effectively isn't doing anything for us? (This goes for other changes as well)